pkg/v21/resource/appmigration/create.go

```diff
@@ -66,17 +67,25 @@ func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
 		LabelSelector: fmt.Sprintf("%s=%s", label.ManagedBy, project.Name()),
 	}
 
+	ctx, cancel := context.WithTimeout(ctx, 3*time.Second)
+	defer cancel()
+
 	chartConfigs, err := tenantG8sClient.CoreV1alpha1().ChartConfigs("giantswarm").List(listOptions)
 	if tenant.IsAPINotAvailable(err) {
 		r.logger.LogCtx(ctx, "level", "debug", "message", "tenant cluster is not available yet")
 		r.logger.LogCtx(ctx, "level", "debug", "message", "canceling resource")
 		resourcecanceledcontext.SetCanceled(ctx)
 		return nil
-	} else if apierrors.IsNotFound(err) {
+	} else if isChartConfigNotInstalled(err) {
```

Comment:
`apierrors.IsNotFound(err)` doesn't work. So applying the fix @tomahawk28 added in https://github.com/giantswarm/cluster-operator/pull/696.